### PR TITLE
Write internallog warning for unrecognized properties used in config on layout renderers

### PR DESCRIPTION
**Pull request by snakefoot**
LayoutParser - InternalLogger Warnings when skipping unrecognized properties for LayoutRenderers

### CODE REVIEW

**Comment by 304NotModified** [comment]
Hi @snakefoot 

Thanks for the PR. Unfortunately I don't fully understand what we here trying to achieve. Does this PR add a new warning to the internal log when something is skipped? What is an example of "something" here?

PS: you could create drafts now. Please do, because I get a notification for every push now (have converted to draft now). If you think it's ready for review, please undraft. Thanks!

**Comment by snakefoot** [comment]
Let's say you spell something wrong in a property-Nam. Like this: tolowercase=true


Then it will not configure anything but just discard the unrecognized property value for the layoutrenderer.


I would like to have a warning when that happens. The PR is done, but playing with build server and exotic target frameworks.




Sent from my Sony Xperia


---- Julian Verdurmen wrote ----


Hi @snakefoot<https://github.com/snakefoot>

Thanks for the PR. Unfortunately I don't fully understand what we here trying to achieve. Does this PR add a new warning to the internal log when something is skipped? What is an example of "something" here?

PS: you could create drafts now. Please do, because I get a notification for every push now (have converted to draft now). If you think it's ready for review, please undraft. Thanks!

**Comment by snakefoot** [comment]
Oops. Had to fix some unnecessary Type-Name-Lookup when doing debug-logging in `SetPropertyFromString`

**Comment by 304NotModified** [comment]
> I would like to have a warning when that happens. The PR is now done.

Cool! I renamed the PR, maybe a bit long. Feel free to rename it (but I prefer to keep it "functional")

**Comment by snakefoot** [comment]
Fine with me. Though it is only for LayoutRenderers

**Comment by snakefoot** [comment]
Looked at the code for assigning default-parameter-value. I guess one is doing this:

`${exception }`

Then one will not get any output, as it would assign space ' ' to the default-parameter `Format`.

File: src/NLog/Internal/SimpleStringReader.cs [hunk 1]
```diff
@@ -123,5 +123,22 @@ internal string Substring(int startIndex, int endIndex)
         {
             return _text.Substring(startIndex, endIndex - startIndex);
         }
+
+        internal string ReadUntilMatch(Func<int, bool> charFinder)
```
Comment (304NotModified) [inline src/NLog/Internal/SimpleStringReader.cs:127]: reuse 👍

File: src/NLog/Layouts/LayoutParser.cs [hunk 2]
```diff
@@ -219,12 +205,24 @@ private static string ParseParameterName(SimpleStringReader sr)
...
+            return simpleValue;
+        }
+
+        private static void ParseParameterUnicodeValue(SimpleStringReader sr, StringBuilder nameBuf)
+        {
+            int ch;
             while ((ch = sr.Peek()) != -1)
             {
-                if (ch == ':' || ch == '}')
+                if (EndOfLayout(ch))
```
Comment (304NotModified) [inline src/NLog/Layouts/LayoutParser.cs:225]: 👍

**Comment by 304NotModified** [comment]
Thanks!

**Pull request merged**